### PR TITLE
Add option to add a newline between 2 adjacent attributes

### LINKED ISSUE #773: Rule for putting annotations into separate lines
We often put various annotations such as `@available` or `@inlinable` in separate lines before methods or types. A rule to enforce that formatting would be great.

```
@usableFromInline
@SomeMacro
struct Foo {
}
```

### PR DESCRIPTION
**Pull request by dduan**
Add an option that inserts hard line breaks between adjacent attributes.

Closes #773

### CODE REVIEW

**Comment by dduan** [comment]
converted draft to investigate whether extra spaces between attributes should be remove and what that means for the naming/implementation of this option.

**Comment by allevato** [comment]
I think we would only want this behavior for _declaration_ attributes, because it would be heavy-handed on type attributes. For example, how does your current implementation handle this case?

```swift
func f(body: @MainActor @Sendable () -> Void) {}
```

We should definitely keep those horizontally flowing, IMO.

Can you add some tests for that case, and also some tests that use attributes in conditional `#if ...` blocks, just to make sure we don't break anything there?

**Comment by dduan** [comment]
makes sense. I'll give it a go!

**Comment by dduan** [comment]
@allevato I took your suggestion and expanded the unit test to cover a lot more node types.

File: Sources/SwiftFormat/API/Configuration.swift [hunk 1]
```diff
@@ -34,6 +34,7 @@ public struct Configuration: Codable, Equatable {
     case lineBreakBeforeControlFlowKeywords
     case lineBreakBeforeEachArgument
     case lineBreakBeforeEachGenericRequirement
+    case lineBreakBetweenAttributes
```
Comment (allevato) [inline Sources/SwiftFormat/API/Configuration.swift:37]: Can we update the name of this to be `lineBreakBetweenDeclarationAttributes` to match the new behavior?

File: Tests/SwiftFormatTests/PrettyPrint/AttributeTests.swift [hunk 2]
```diff
@@ -468,4 +468,103 @@ final class AttributeTests: PrettyPrintTestCase {
 
     assertPrettyPrintEqual(input: input, expected: expected, linelength: 100)
   }
+
+  func testLineBreakBetweenAttributes() {
```
Comment (allevato) [inline Tests/SwiftFormatTests/PrettyPrint/AttributeTests.swift:472]: Can you add a test that includes attributes surrounded by `#if`?

I would expect

```swift
@foo @foo2
#if FLAG
  @bar @baz
#endif
@quux func f() {}
```

to become

```swift
@foo
@foo2
#if FLAG
  @bar
  @baz
#endif
@quux
func f() {}
```

Comment (dduan) [reply Tests/SwiftFormatTests/PrettyPrint/AttributeTests.swift:472]: yup

**Comment by dduan** [comment]
@allevato thanks for the feedbacks! I've addressed them.

**Comment by allevato** [comment]
Thanks!

**Pull request merged**